File: 1 - Algorithmic Toolbox/Week_2/6 - Last Digit of the Sum of Fibonacci Numbers/last_digit_of_sum_fibonacci_numbers.py

def sum_of_fibonacci( fibonacci_index: int ) -> int:
    fibonacci_index %= 60
    
    if ( fibonacci_index <= 1 ):
        return fibonacci_index

    total: int = 1
    prev_val: int = 0
    curr_val = 1
    temp_val: int = 0

    for i in range( 1, fibonacci_index ):
        temp_val = prev_val
        prev_val = curr_val
        curr_val = ( temp_val + prev_val ) % 10
        total += curr_val

    return total % 10


def main( ) -> None:
    user_input = int( input( ) )
    print( sum_of_fibonacci( user_input ) )

if __name__ == "__main__":
    main( )

File: 1 - Algorithmic Toolbox/2 - Algorithmic Warmup/6 - Last Digit of the Sum of Fibonacci Numbers/last_digit_of_sum_fibonacci_numbers.c
#include <stdio.h>

long sum_of_fibonacci(long fibonacci_index) 
{
    fibonacci_index %= 60 ;

    if (fibonacci_index <= 1) {
        return fibonacci_index ;
    }

    long total = 1 ;
    long prev_val = 0 ;
    long curr_val = 1 ;
    long temp_val = 0 ;

    for (long i = 1; i < fibonacci_index; i++) {
        temp_val = prev_val ;
        prev_val = curr_val ;
        curr_val = (temp_val + prev_val) % 10 ;
        total += curr_val ;
    }

    return total % 10 ;
}

int main(void) 
{
    long user_input ;
    scanf("%ld", &user_input) ;
    printf("%ld\n", sum_of_fibonacci(user_input)) ;

    return 0 ;
}